components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt

```diff
@@ -266,8 +266,10 @@ class CrashReporter(
             // If the app has not registered any intent then we can't send one.
             false
         } else {
-            // If this is a non-fatal native code crash then we can recover and can notify the app
-            crash is Crash.NativeCodeCrash && !crash.isFatal
+            // If this is a native code crash in a foreground child process then we can recover
+            // and can notify the app. Background child process crashes will be recovered from
+            // automatically, and main process crashes cannot be recovered from.
+            crash is Crash.NativeCodeCrash && crash.processType == Crash.NativeCodeCrash.PROCESS_TYPE_FOREGROUND_CHILD
```

Comment:
Do you mean `crash.processType != Crash.NativeCodeCrash.PROCESS_TYPE_MAIN`.  

There's a lot of place we're checking if it's fatal.  Is is another good reason to keep the `isFatal` property so we don't have to update these checks.

Reply:
In this case I actually do mean `crash.processType == Crash.NativeCodeCrash.PROCESS_TYPE_FOREGROUND_CHILD`.

Agi suggested that we use the crash notification for background child crashes, which as I understand it is handled in android-components rather than by the application via the intent. So we only want to send intent for FOREGROUND_CHILD crashes.

That does seem fairly arbitrary, though. Perhaps it would be better if we send the intent for all non-fatal crashes (both FOREGROUND_CHILD and BACKGROUND_CHILD). And if we expose a public API for showng the crash notification, then Fenix's intent handler could call that API in the BACKGROUND_CHILD case?

That would seem more consistent to me, what do you think?

Reply:
Ah, I see.  It does make sense to only notify the app side when it's a FOREGROUND_CHILD crash.  I'm OK with how you've implemented it.  Thanks